### PR TITLE
ci: split logictest into two parts base and ydb

### LINKED ISSUE #6697: SQL logic test is too slow that exceeds 10 minutes limitation
SQL logic test is too slow that exceeds 10 minutes limitation.

We can extend this limitation to 20 or 30 minutes, but we should improve this in the future.

I expect SQL logic tests could be finished within 10 minutes (maybe we can execute them concurrently)?

_Originally posted by @Xuanwo in https://github.com/datafuselabs/databend/issues/6681#issuecomment-1189028385_

### PR DESCRIPTION
**Pull request by ZeaLoVe**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Split logictest into two parts base(original stateless suites) and ydb. Seems the debug build need more time to finish tests.
This pr will decrease test time by adding more logictest job. Make sure each test job run less than 10min.

Fixes #6697

### CODE REVIEW

**Comment by Xuanwo** [comment]
Please use github action matrix instead of repeating similiar workflow.

**Comment by ZeaLoVe** [comment]
> Please use github action matrix instead of repeating similiar workflow.

PTAL

File: .github/workflows/dev-linux.yml [hunk 1]
```diff
@@ -121,12 +121,19 @@ jobs:
       - uses: ./.github/actions/test_stateless_cluster_linux
 
   test_sqllogic_standalone_linux:
-    timeout-minutes: 20
+    timeout-minutes: 15
```
Comment (Xuanwo) [inline .github/workflows/dev-linux.yml:124]: Assign a good understanding name like:

```yaml
name: test_sqllogic_standalone_linux_${{ matrix.dirs }}
```

**Comment by Xuanwo** [comment]
Please also update `mergify.yml` so that our auto merge still works.

File: .github/workflows/dev-linux.yml (hunk 1 above)
Comment (ZeaLoVe) [reply .github/workflows/dev-linux.yml:124]: Can this name contain a space? For example "test_sqllogic_standalone_linux_gen query" ?

Comment (everpcpc) [reply .github/workflows/dev-linux.yml:124]: why not make it single dir per job?

Comment (Xuanwo) [reply .github/workflows/dev-linux.yml:124]: > Can this name contain a space? For example "test_sqllogic_standalone_linux_gen query" ?

I don't think so.

Comment (ZeaLoVe) [reply .github/workflows/dev-linux.yml:124]: Historical Reason, I can move query into gen, and rename gen to databend, Maybe Clearer.

Comment (everpcpc) [reply .github/workflows/dev-linux.yml:124]: > Can this name contain a space? For example "test_sqllogic_standalone_linux_gen query" ?

Yes, it is OK. But not beautiful.

**Comment by ZeaLoVe** [comment]
@everpcpc @Xuanwo  PTAL

**Comment by Xuanwo** [comment]
Visit https://patch-diff.githubusercontent.com/raw/datafuselabs/databend/pull/6965.diff for diff if Github's web UI is broken.

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
If you can't approve this PR, please use `gh pr review --approve 6965`

**Review by everpcpc — APPROVED** [review]

**Pull request merged**